### PR TITLE
[Pillar] Design guidelines

### PR DESCRIPTION
**Pull request by josepmartins**
## Summary

Add `Pillar` design guidelines.

## List of notable changes:

- **added** new ComponentLayout to `Pillar` docs
- **added** design guidelines

## What should reviewers focus on?

- 👀 Documentation: 

## Supporting resources (related issues, external links, etc):

- Epic: https://github.com/github/primer/issues/2231
- Tracking issue: https://github.com/github/primer/issues/2129

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

 </td>
<td valign="top">

</td>
</tr>
</table>

### CODE REVIEW

File: apps/docs/content/components/Pillar/index.mdx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+---
+
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+## Anatomy
+
+![An image showing the Pillar anatomy with an icon, a heading and a description in a vertical order.](https://github.com/primer/brand/assets/912236/c52debd2-4277-42c9-9a73-bd6bd087dd1e)
+
+- Icon: It provides a visual cue to the user about the content of the piller. Icon is optional.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:13]:
```suggestion
- Icon: An optional visual cue related to the content of the Pillar.
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+- Heading: The main title of the pillar. It should be short and concise.
```
Comment (rezrah) [inline apps/docs/content/components/Pillar/index.mdx:14]: Should pillar be capitalized as it's the component name? We do this with all other components, like Grid below so I think we should do that here too.

```suggestion
- Heading: The main title of the Pillar. It should be short and concise.
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+- Description: Short text that extends the information provided by the heading.
+
+## Usage
+
+Use pillars to group similar features or benefits together to easily scan and read. Pillars displays content in a familiar and recognizable style.
```
Comment (danielguillan) [inline apps/docs/content/components/Pillar/index.mdx:19]:
```suggestion
Use pillars to group similar features or benefits together to easily scan and read. Pillars display content in a familiar and recognizable style.
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      Use icons either for all pillars or none for a visually balanced layout.
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/912236/8ddb7a7b-396b-44c2-9cbd-e6a44eb13710" />
+    <Caption>Don't use icons sparingly.</Caption>
+  </Dont>
+</DoDontContainer>
+
+When presenting numerous pillars, we recommend stacking them vertically. You can use the [Grid](/components/Grid) or the [Stack](/components/Stack) component to achieve this. We recommend using a grid with 3 or 4 columns.
```
Comment (danielguillan) [inline apps/docs/content/components/Pillar/index.mdx:55]:
```suggestion
When presenting more than four pillars, we recommend displaying them in multiple rows. You can use the [Grid](/components/Grid) or the [Stack](/components/Stack) component to achieve this. We recommend using a grid with 3 or 4 columns and balancing the number of pillars on each row.
```

File: apps/docs/content/components/Pillar/index.mdx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+![An image showing the Pillar anatomy with an icon, a heading and a description in a vertical order.](https://github.com/primer/brand/assets/912236/1d031ae9-bf4e-4e19-a7b7-c8cc0b7e5940)
+
+## Options
+
+### Align
+
+Pillars can be aligned to the start or center of the container. Aligning to the start is the default and recommended behavior. Use center alignment when in groups of 3 or 4 pillars and if their title and descriptions are short, so that the content is visually balanced and easy to scan.
```
Comment (danielguillan) [inline apps/docs/content/components/Pillar/index.mdx:63]: I think we need a bit more guidance as to when center alignment can be used.

**Review by danielguillan — APPROVED** [review]

**Pull request merged**